test/common/CollectiveArgs.hpp

```diff
@@ -72,6 +74,22 @@ namespace RcclUnitTesting
     char ptr[MAX_RANKS * sizeof(double)];
   };
 
+  struct OptionalColArgs
+  {
+    int      collId = -1;
+    int      rank = -1;
+    PtrUnion scalarsPerRank = {nullptr};
+    int      scalarMode = -1;
+    // allToAllv args
+    size_t   sendcounts[MAX_RANKS*MAX_RANKS];
+    size_t   sdispls[MAX_RANKS*MAX_RANKS];
+    size_t   recvcounts[MAX_RANKS*MAX_RANKS];
+    size_t   rdispls[MAX_RANKS*MAX_RANKS];
+    size_t   numInputElementsArray[MAX_RANKS];
+    size_t   numOutputElementsArray[MAX_RANKS];
```

Comment:
These need to be renamed or at least commented to match naming styles in the rest of the code.  It might not be immediately obvious what sdispls is.  
Also, why not just use 2D arrays? Is it only because these are passed directly into alltoallv?